ocdata/core/adslab.py

```diff
@@ -0,0 +1,271 @@
+import ase
+import numpy as np
+import scipy
+from itertools import product
+
+from ocdata.core import Adsorbate, Surface
+from ase.data import atomic_numbers, covalent_radii
+import copy
+
+
+class Adslab:
+    """
+    Initializes an adsorbate-catalyst system for a given Adsorbate and Surface.
+
+    Arguments
+    ---------
+    surface: Surface
+        Surface object.
+    adsorbate: Adsorbate
+        Adsorbate object.
+    num_sites: int
+        Number of sites to sample.
+    num_augmentations_per_site: int
+        Number of augmentations of the adsorbate per site. Total number of
+        generated structures will be `num_sites` * `num_augmentations_per_site`.
+    height_adjustment: float
+        Distance in Angstroms to add to the height of placement incrementally
+        until there is no atomic overlap.
+    """
+
+    def __init__(
+        self,
+        surface: Surface,
+        adsorbate: Adsorbate,
+        num_sites: int = 100,
+        num_augmentations_per_site: int = 1,
+        height_adjustment: float = 0.1,
+    ):
+        self.surface = surface
+        self.adsorbate = adsorbate
+        self.num_sites = num_sites
+        self.num_augmentations_per_site = num_augmentations_per_site
+
+        self.sites = self.get_binding_sites(num_sites)
+        self.structures = self.place_adsorbate_on_sites(
```

Comment:
good to align the naming here with the class once we decide on a name for the class. structure makes me think of pymatgen. Maybe split the `(site, structure)` tuple so this may be stored in `Adslab.atoms` to better parallel the other classes (i.e. `Bulk`, `Adsorbate`, `Surface`)

Reply:
From the last discussion we had on this — we'll probably rename this class to `AdsorbateSurfaceConfig` or `AdsorbateSlabConfig`, and rename the `Surface` class to `Slab`. `get_binding_sites` probably belongs in the `Slab` class, not here, so I might move that.

> Maybe split the (site, structure) tuple so this may be stored in Adslab.atoms 

Yeah.. it's still a bit different from the other classes because `Adslab.atoms` would be storing a list of ase.Atoms objects, not just one. But I don't have a better naming proposal..

Reply:
Updated naming: https://github.com/Open-Catalyst-Project/Open-Catalyst-Dataset/blob/refactor-v2/ocdata/core/adsorbate_slab_config.py#L52-L53